modular_robot_physical/revolve2/modular_robot_physical/_physical_robot_controller.py

```diff
@@ -0,0 +1,154 @@
+import argparse
+import asyncio
+import sys
+import time
+from dataclasses import dataclass
+
+from ._physical_robot_config import PhysicalRobotConfig
+from .physical_robot_control_interfaces import (
+    Pca9685PhysicalControlInterface,
+    PhysicalControlInterface,
+    V1PhysicalControlInterface,
+)
+from .physical_robot_sensor_states import PhysicalSensorState
+
+
+@dataclass
+class _LogEntry:
+    timestamp: int
+
+
+class PhysicalRobotController:
+    """Controller for the physical Robot."""
+
+    _config: PhysicalRobotConfig
+    _hardware_interface: PhysicalControlInterface
+    _sensor_interface: PhysicalSensorState
+    _stop: bool
+    _log: list[_LogEntry]
+
+    _debug: bool
+    _log_file: str | None
+
+    _control_period: float
+
+    def __init__(self) -> None:
+        """Initialize the Controller."""
+        self._stop = False
+
+    def main(self) -> None:
+        """
+        Execute the controller for the physical robot.
+
+        :raises NotImplementedError: If for a certain robot type no physical control interface is defined.
+        """
+        try:
+            parser = argparse.ArgumentParser()
+            parser.add_argument(
+                "physical_robot_config"  # TODO: add type once set
+            )  # os.fsencode is bytes  mybe think of other way
+            parser.add_argument("--hardware", type=str)
```

Comment:
you can require the user to choose from a list of options. I think it's called `choices`